### PR TITLE
Housekeeping: python package deps upgrade

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- twilio upgrade: https://github.com/RasaHQ/rasa/pull/12583
- aiohttp upgrade: https://github.com/RasaHQ/rasa/pull/12690
- certifi upgrade: https://github.com/RasaHQ/rasa/pull/12700
- matplotlib: https://github.com/RasaHQ/rasa/pull/12629
- tarsafe: https://github.com/RasaHQ/rasa/pull/12525
- rocketchat-api: https://github.com/RasaHQ/rasa/pull/12402

not upgrading ruff, created a JIRA ticket instead: https://rasahq.atlassian.net/browse/ENG-440

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -301,7 +301,7 @@ git = "https://github.com/RasaHQ/pytest-sanic"
 branch = "fix_signal_issue"
 
 [tool.poetry.group.dev.dependencies]
-ruff = ">=0.0.255,<0.0.256"
+ruff = ">=0.0.255,<0.0.277"
```
Comment (ancalita) [inline pyproject.toml:304]: Please fix any new errors found by `ruff` as part of this upgrade.

File: pyproject.toml [hunk 2]
```diff
@@ -331,7 +331,7 @@ types-setuptools = "^67.2.0"
 memory-profiler = "^0.61.0"
 psutil = "^5.8.0"
 mypy-extensions = "^0.4.3"
-sanic-testing = ">=21.12.0,<22.9.0"
+sanic-testing = ">=21.12.0,<23.4.0"
```
Comment (ancalita) [inline pyproject.toml:334]: There are a lots of tests failing I think because of this upgrade, please check that sanic-testing is compatible with the version of sanic we have pinned, if not don't try to upgrade, the major.minor version need to be aligned for these dependencies.

Comment (ancalita) [reply pyproject.toml:334]: Also as reference for the future, if you upgrade a dependency and it breaks tests, as part of the PR you will have to fix these breaking changes and adapt code.

**Comment by vcidst** [comment]
@varunshankar I've bumped pyyaml to >6.0 in #12678 which seems to address the CI errors

File: pyproject.toml (hunk 1 above)
Comment (vcidst) [reply pyproject.toml:304]: Ruff upgrade opens a whole new can of worms,
```
Found 68 errors.
[*] 53 potentially fixable with the --fix option.
```

I think that we can skip upgrading ruff in this PR and instead create a ticket to do it later

File: .github/scripts/download_pretrained.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def get_model_name_and_weights_from_config(
             if model_name not in model_class_dict:
                 raise KeyError(
                     f"'{model_name}' not a valid model name. Choose from "
-                    f"{str(list(model_class_dict.keys()))} or create"
+                    f"{list(model_class_dict.keys())!s} or create"
```
Comment (ancalita) [inline .github/scripts/download_pretrained.py:55]: Why was this change required?

**Review by ancalita — APPROVED** [review]
Left only 1 question to be addressed, otherwise LGTM 👍🏻

File: .github/scripts/download_pretrained.py (hunk 3 above)
Comment (varunshankar) [reply .github/scripts/download_pretrained.py]: Oh my bad, that was fixed automatically by ruff, missed this when reverting the ruff upgrade. Thanks
